lib/shared/sdk/usbboot.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This work is heavily based on https://github.com/raspberrypi/usbboot
+ * Copyright 2016 Raspberry Pi Foundation
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const Bluebird = require('bluebird')
+const usb = require('usb')
+
+/**
+ * @summary The number of USB interfaces in devices with a BCM2837 SoC
+ * @description
+ * Raspberry Pi 3, and CM3 use this SoC.
+ * @type {Number}
+ * @constant
+ */
+const INTERFACES_SOC_BCM2837 = 1
+
+/**
+ * @summary Vendor ID of "Broadcom Corporation"
+ * @type {Number}
+ * @constant
+ */
+const USB_VENDOR_ID_BROADCOM_CORPORATION = 0x0a5c
+
+/**
+ * @summary List of usbboot capable devices
+ * @type {Object[]}
+ * @constant
+ */
+const USBBOOT_CAPABLE_USB_DEVICES = [
+
+  // Raspberry Pi 1
```

Comment:
Might be nicer to refer to just the SoC (i.e. `BCM2835`) rather than `Raspberry Pi 1`, since the latter implies a particular board? Whereas the same SoC is shared across multiple boards. (and as far as usb-booting is concerned, it's only the SoC that matters)